### PR TITLE
Implementation of sessionBrowserName and a fix for maxSessions (#2709 and #3061

### LINKED ISSUE #2709: Selenium Gird Scaler | Edge active sessions not being properly counted
### Report

Been working on getting a selenium grid 4 deployment working in AKS using Keda to scale. Chrome and Firefox have been no problem but scaling Edge browser pods seems to have a bug.

Currently using the following scaled-object:
```
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: selenium-edge-scaledobject
  namespace: legion-selenium-grid
  labels:
    deploymentName: selenium-edge-node-deployment
spec:
  minReplicaCount: 0
  maxReplicaCount: 80
  scaleTargetRef:
    name: selenium-edge-node-deployment
  triggers:
    - type: selenium-grid
      metadata:
        url: 'https://selenium-grid-qe-tools.accruentsystems.com/graphql'
        browserName: 'MicrosoftEdge'
```

New sessions in the queue appear as 'MicrosoftEdge' and scale up just fine:
image

However, taking a look at the active sessions I noticed that the browserName on the active sessions becomes 'msedge':
image

Thus when it comes time for Keda to check for a scale-down event it sees that there are no active sessions and proceeds scale to 0.

### Expected Behavior

Keda detects that Edge pods are still in use and scales down gradually as needed.

### Actual Behavior

When scaling down Keda will remove all Edge pods regardless of current active sessions.

### Steps to Reproduce the Problem

1. Deploy selenium grid and Edge deployments
2. Apply scaled-object for edge (see description)
3. Can be done in a variety of ways but an easy way: Run a single long (10 minute +) test on selenium grid. Grid will scale up to allow your test to run but will eventually scale to 0 before your test can finish.

### Logs from KEDA operator

```
example
```

### KEDA Version

2.6.1

### Kubernetes Version

1.21

### Platform

Microsoft Azure

### Scaler Details

Selenium Grid

### Anything else?

I am not in any way familiar with Go and am having issues deploying a local change to AKS but I believe all that really needs done here is a specific check for "MicrosoftEdge" as a browserName and then simply use the session-specific naming of 'msedge' in that situation.

Probably something like this (though im sure there is a more elegant way to put it). [Around line 225 in selenium_grid_scaler](https://github.com/kedacore/keda/blob/main/pkg/scalers/selenium_grid_scaler.go#L225):
```
var sessions = seleniumResponse.Data.SessionsInfo.Sessions
	for _, session := range sessions {
		var capability = capability{}
		var browserCapName = ""
		if browserName == "MicrosoftEdge" {
			browserCapName = "msedge"
		} else {
			browserCapName = browserName
		}
		if err := json.Unmarshal([]byte(session.Capabilities), &capability); err == nil {
			if capability.BrowserName == browserCapName {
				if strings.HasPrefix(capability.BrowserVersion, browserVersion) {
					count++
				} else if browserVersion == DefaultBrowserVersion {
					count++
				}
			}
		} else {
			seleniumGridLog.Error(err, fmt.Sprintf("Error when unmarshaling sessions info: %s", err))
		}
	}
```

### LINKED ISSUE #3061: Selenium Grid Scaler | Max Sessions implementation issue
### Report

The MaxSessions implementation in #2774 is producing incorrect results. I ran into this while trying to create a fix for #2709.

### Expected Behavior

We need to take into consideration the count of the nodes we have in this calculation.

Something like:
`count = count / (gridMaxSession / nodeCount)`

1. 2 sessions running on 2 nodes, 3 sessions in the queue, each node accepts 1 session max:
   a. count = 5 / (2 / 2)
   b. This equals out to be 5 which is how many nodes we need.
2. 10 sessions running on 2 nodes, 7 sessions in the queue, each node accepts 5 sessions max: 
   a. count = 17 / (10 / 2)
   b. This equals out to be 3.4 (rounded to 4 I assume?) which is what we need.

### Actual Behavior

https://github.com/kedacore/keda/blob/main/pkg/scalers/selenium_grid_scaler.go#L252 :
`count = (count + gridMaxSession - 1) / gridMaxSession` is the current equation.

This equation does not take into account the number of nodes available and thus will consistently produce an incorrect result, for example:

1. 2 sessions running on 2 nodes, 3 sessions in the queue, each node accepts 1 session max:
   a. count = (5 + 2 - 1) / 2
   b. This equals out to be 3 when it should be 5
2. 10 sessions running on 2 nodes, 7 sessions in the queue, each node accepts 5 sessions max: 
   a. count = (17 + 10 - 1) / 10
   b. This equals out to be 2.6 (rounded to 3 I assume?) when it should be 4

### Steps to Reproduce the Problem

Simply using the scaler with a version higher than 2.6.1 will suffice. Items will be left in the queue while we still have room to scale.

### Logs from KEDA operator

_No response_

### KEDA Version

2.7.1

### Kubernetes Version

1.23

### Platform

Microsoft Azure

### Scaler Details

Selenium Grid

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Wolfe1**
1. Fixed the implementation of MaxSessions to now give an accurate count of nodes to scale up or down.
2. Implemented an optional parameter `sessionBrowserName` to allow users to specify a different name for their browser in the active sessions list. In this case this is the behavior of Edge where in the session queue it is called "MicrosoftEdge" and as an active session it is called "msedge".

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes #2709 
Fixes #3061 

Docs PR: https://github.com/kedacore/keda-docs/pull/764

### CODE REVIEW

**Review by JorTurFer — COMMENTED** [review]
In general, looks good. Thanks for these improvements.
Only 2 things:

- Unit tests are failing (Test_getCountFromSeleniumResponse)
- Please, add an e2e test case covering this change (adding an edge node)

**Review by zroubalik — COMMENTED** [review]
Thanks! Next time please open a separate PR for each change, it is easier to review and maintain then.

**Comment by JorTurFer** [comment]
hey @Wolfe1 
e2e tests are failing, I think because the current e2e image doesn't enqueue any job in edge nodes. Could you update it?
[It's in the repository test-tools](https://github.com/kedacore/test-tools/tree/main/e2e/images/selenium-grid) 
TBH, I have no idea about how to update it :(

**Comment by Wolfe1** [comment]
@JorTurFer Really giving me the full tour of the keda repos haha

**Comment by JorTurFer** [comment]
> @JorTurFer Really giving me the full tour of the keda repos haha

yes, and I only can say one thing,  THANKS for the huge effort you are doing ❤️ ❤️ ❤️

**Comment by Wolfe1** [comment]
@JorTurFer Haha no worry, enjoying it.

Still working through these unit tests, not quite sure why I am getting the results I am but at least I figured out how to run them locally now so you should get less spam from me.

**Comment by JorTurFer** [comment]
> Still working through these unit tests, not quite sure why I am getting the results I am but at least I figured out how to run them locally now so you should get less spam from me.

Don't worry, I can show you how to do it locally xD. The easiest way is executing from root folder `make test` but that will run all unit tests. Most probably you prefer to run/debug only your tests to reduce the execution time and troubleshoot the issues, if you are using Visual Studio Code with the Golang extensions, you can just click on the contextual buttons:
image

File: tests/scalers/selenium-grid.test.ts [hunk 1]
```diff
@@ -144,15 +154,17 @@ test.serial('should create two chrome and one firefox nodes', t => {
...
     seleniumFireFoxReplicaCount = seleniumFireFoxReplicaCount != '1' ? sh.exec(`kubectl get deploy/selenium-firefox-node -n ${seleniumGridNamespace} -o jsonpath='{.spec.replicas}'`).stdout : seleniumFireFoxReplicaCount;
+    seleniumEdgeReplicaCount = seleniumEdgeReplicaCount != '1' ? sh.exec(`kubectl get deploy/selenium-edge-node -n ${seleniumGridNamespace} -o jsonpath='{.spec.replicas}'`).stdout : seleniumEdgeReplicaCount;
     seleniumChrome91NodeReplicaCount = seleniumChrome91NodeReplicaCount != '1' ? sh.exec(`kubectl get deploy/selenium-chrome-node-91 -n ${seleniumGridNamespace} -o jsonpath='{.spec.replicas}'`).stdout : seleniumChrome91NodeReplicaCount;
-    if (seleniumChromeNodeReplicaCount == '1' && seleniumFireFoxReplicaCount == '1' && seleniumChrome91NodeReplicaCount == '1') {
+    if (seleniumChromeNodeReplicaCount == '1' && seleniumFireFoxReplicaCount == '1' && seleniumEdgeReplicaCount == '1' && seleniumChrome91NodeReplicaCount == '1') {
       break;
     }
-    console.log('Waiting for chrome to scale up 2 pods and firefox to 1 pod');
+    console.log('Waiting for chrome to scale up 2 pods, firefox to 1 pod, and edhe to 1 pod');
```
Comment (JorTurFer) [inline tests/scalers/selenium-grid.test.ts:167]: I have discovered a typo here, `edhe` xD

Comment (Wolfe1) [reply tests/scalers/selenium-grid.test.ts]: Fixed, thanks!

**Comment by Wolfe1** [comment]
Alright @JorTurFer I think we are ready for one last e2e run and it should be good to review

**Review by JorTurFer — APPROVED** [review]
LGTM!

**Comment by JorTurFer** [comment]
Thanks a lot for these improvements! ❤️ 
Let's wait till e2e tests finish and I'll merge it then
